File: solutions/isbn-verifier/Python/isbn_verifier.py
import re

def is_valid(isbn):
    isbn = isbn.replace("-", "")
    if re.search(r"[\d]{9}[\dX]$", isbn) is None or len(isbn) != 10:
        return False

    product = sum(factor * (10 if d == 'X' else int(d)) for factor, d in zip(range(10,0,-1), isbn))

    return product % 11 == 0

File: solutions/isbn-verifier/C++/isbn_verifier.cpp
#include "isbn_verifier.h"

namespace isbn_verifier {
    bool is_valid(const std::string_view& isbn) {
        auto isbn10 = 0, factor = 10;
        for (char c: isbn) {
            if (c == '-')
                continue;
            bool digit_is_valid = ('0' <= c && c <= '9') || (factor == 1 && c == 'X');
            if (factor < 1 || digit_is_valid == false)
                return false;
            int digit = c == 'X' ? 10 : c - '0';
            isbn10 += digit * factor--;
        }
        return factor == 0 && isbn10 % 11 == 0;
    }
}
